Prox/Prox/Utilities/PlaceUtilities.swift

```diff
@@ -21,6 +22,43 @@ struct PlaceUtilities {
         }
     }
 
+    static func sort(places: [Place], byTravelTimeFromLocation location: CLLocation, ascending: Bool = true, completion: @escaping ([Place]) -> ()) {
+        PlaceUtilities.getTravelTimes(forPlaces: places, fromLocation: location).upon { result in
+            let sortedPlaces = places.sorted { (placeA, placeB) -> Bool in
+                guard let placeATravelTimes = PlaceUtilities.lastTravelTimes(forPlace: placeA),
+                    let placeBTravelTimes = PlaceUtilities.lastTravelTimes(forPlace: placeB) else {
+                        return false
+                }
+
+                let placeAETA = placeATravelTimes.getShortestTravelTime()
+                let placeBETA = placeBTravelTimes.getShortestTravelTime()
+
+                guard ascending else {
+                    return placeAETA > placeBETA
+                }
+                return placeAETA < placeBETA
+            }
+            completion(sortedPlaces)
+        }
+    }
+
+    private static func getTravelTimes(forPlaces places: [Place], fromLocation location: CLLocation) -> Future<[DatabaseResult<TravelTimes>]> {
+        var placesWithTimes = [Deferred<DatabaseResult<TravelTimes>>]()
+        for place in places {
+            let placeDeferred = place.travelTimes(fromLocation: location)
+            placesWithTimes.append(placeDeferred)
+        }
+        return placesWithTimes.allFilled()
```

Comment:
Does this always complete? I was having issues with the travel time UI occasionally only showing a spinner when scrolling quickly – I don't know if this was a UI bug or a logic bug, but I vaguely remember it being the `TravelTimesProvider` method not returning.